actioncable/lib/action_cable/subscription_adapter/redis.rb

```diff
@@ -0,0 +1,45 @@
+begin
+  require 'em-hiredis'
+  require 'redis'
+rescue Gem::LoadError => e
+  raise Gem::LoadError, "You are trying to use the Redis Action Cable adapter, but do not have the proper gems installed. Add `gem 'em-hiredis'` and `gem 'redis'` to your Gemfile (and ensure its version is at the minimum required by Action Cable)."
+end
+
+module ActionCable
+  module SubscriptionAdapter
+    class Redis < Base
+      # The redis instance used for broadcasting. Not intended for direct user use.
+      def broadcast(channel, payload)
+        broadcast_redis_connection.publish(channel, payload)
+      end
+
+      def subscribe(channel, callback, success_callback = nil)
```

Comment:
Since there's a variable called `success_callback` that leaves `callback`'s intent up in the air. Starting to question the suffix entirely too. Perhaps the more succinct `subscriber` and `notifier`, or `on_subscribe` and `on_success` could capture what you're trying to convey.